app/assets/javascripts/application/config.js

```diff
@@ -0,0 +1,63 @@
+(function (angular) {
+  'use strict';
+
+  angular
+      .module('application')
+      .config(applicationConfig)
+      .config(configureAuth);
+
+  /* @ngInject */
+  function applicationConfig($httpProvider, $routeProvider) {
+    $httpProvider.interceptors.push('AuthInterceptor');
+
+    $routeProvider.otherwise({redirectTo: '/'});
+  }
+
+  /* @ngInject */
+  function configureAuth($authProvider) {
+    $authProvider.configure({
+      apiUrl: window.location.protocol + '//' + window.location.host,
+      tokenValidationPath: '/auth/validate_token',
+      signOutUrl: '/auth/sign_out',
+      emailRegistrationPath: '/auth',
+      accountUpdatePath: '/auth',
+      accountDeletePath: '/auth',
+      confirmationSuccessUrl: window.location.href,
+      passwordResetPath: '/auth/password',
+      passwordUpdatePath: '/auth/password',
+      passwordResetSuccessUrl: window.location.href,
+      emailSignInPath: '/auth/sign_in',
+      storage: 'localStorage',
```

Comment:
@tsov using localStorage here as the cookie store implementation appeared to have some strange race conditions, resulting in 401 for no reason